app/models/miq_server.rb

```diff
@@ -369,6 +369,13 @@ def monitor_loop
       _log.info "Server Monitoring Complete - Timings: #{timings.inspect}" unless timings[:total_time] < server_log_timings_threshold
       sleep monitor_poll
     end
+  rescue Interrupt => e
+    _log.info("Received #{e.message} signal, killing server")
+    self.class.kill
+    exit 1
+  rescue SignalException => e
```

Comment:
Note, we're now handling other soft signals I believe.  I don't recall why we were only handling term, usr1, and usr2 as soft signals before.  Maybe it doesn't matter?